baremetal/metal3machine_manager.go

```diff
@@ -808,13 +808,23 @@ func (m *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost, *p
 		default:
 			continue
 		}
+
 		// continue if BaremetalHost is paused
 		annotations := host.GetAnnotations()
 		if annotations != nil {
 			if _, ok := annotations[bmh.PausedAnnotation]; ok {
 				continue
 			}
 		}
+
+		// continue if BaremetalHost is unhealthy
+		annotations := host.GetAnnotations()
```

Comment:
annotations is already delared in line 813 isn't it? We don't need this line I suppose, we can even take the whole `if _, ok := annotations[bmh.UnhealthyAnnotation]; ok`  inside the if defined in line 814

Reply:
Good point, better to move it as part of the previous annotation check.